moabb/datasets/Dreyer2023.py

```diff
@@ -0,0 +1,307 @@
+"""
+A large EEG right-left hand motor imagery dataset.
+It is organized into three A, B, C datasets.
+https://zenodo.org/record/7554429
+"""
+
+import logging
+import os
+import shutil
+import zipfile
+from functools import partialmethod
+from os.path import dirname, join
+from pathlib import Path
+
+import mne
+import pandas as pd
+from mne.channels import make_standard_montage
+from mne.io import read_raw_gdf
+from pooch import Unzip, retrieve
+
+from moabb.datasets import download as dl
+
+from .base import BaseDataset
+from .download import get_dataset_path
+
+
+DREYER2023_URL = "https://zenodo.org/record/7554429/files/BCI Database.zip"
+
+SUBJ_INFO = {
+    "Demo_Bio": [
+        "SUJ_gender",
+        "Birth_year",
+        "Vision",
+        "Vision_assistance",
+        "Symptoms_TXT",
+        "Level of study",
+        "Level_knowledge neuro",
+        "Meditation practice",
+        "Laterality answered",
+        "Manual activity",
+        "Manual activity TXT",
+    ],
+    "OpenVibe_Perf": ["Perf_RUN_3", "Perf_RUN_4", "Perf_RUN_5", "Perf_RUN_6"],
+    "Mental_Rotation": ["score", "time_1", "time_2"],
+    "PRE_session": [
+        "PRE_Mood",
+        "PRE_Mindfulness",
+        "PRE_Motivation",
+        "PRE_Hours_sleep_last_night",
+        "PRE_Usual_sleep",
+        "PRE_Level_of_alertness",
+        "PRE_Stimulant_doses_12h",
+        "PRE_Stimulant_doses_2h",
+        "PRE_Stim_normal",
+        "PRE_Tabacco",
+        "PRE_Tabacco_normal",
+        "PRE_Alcohol",
+        "PRE_Last_meal",
+        "PRE_Last_pills",
+        "PRE_Pills_TXT",
+        "PRE_Nervousness",
+        "PRE_Awakening",
+        "PRE_Concentration",
+    ],
+    "POST_session": [
+        "POST_Mood",
+        "POST_Mindfulness",
+        "POST_Motivation",
+        "POST_Cognitive load",
+        "POST_Agentivity",
+        "POST_Expectations_filled",
+    ],
+    "Index_of_Learnig_Style": [
+        "active",
+        "reflexive",
+        "sensory",
+        "intuitive",
+        "visual",
+        "verbal",
+        "sequential",
+        "global",
+    ],
+    "16Pf5": [
+        "A",
+        "B",
+        "C_",
+        "E",
+        "F",
+        "G",
+        "H",
+        "I",
+        "L",
+        "M",
+        "N",
+        "O",
+        "Q1",
+        "Q2",
+        "Q3",
+        "Q4",
+        "IM",
+        "EX",
+        "AX",
+        "TM",
+        "IN",
+        "SC",
+        "Interrogation",
+    ],
+}
+
+
+class Dreyer2023(BaseDataset):
```

Comment:
You may rename this class as `Dreyer2023Base` to make clear that is the parent class and you could indicate in the docstring that it should not be used directly, instead use `Dreyer2023A`, Dreyer2023B` or `Dreyer2023C`.